### PR TITLE
Sync formik input

### LINKED ISSUE #53: [Syncing] formik/Input
## Points to note
- Component in action - `formik/Input.js`
- Github Urls to this component:
  * [neeto-ui/formik/Input.js](https://github.com/bigbinary/neeto-ui/tree/67ee4a5188c5feccbeca2767107651fb24932764/lib/components/formik/Input.js)
  * [NitroUI/Formik/Input.js](https://github.com/bigbinary/neeto-help-web/tree/0b11a7b0f98ac7c533df1e9ec37cf900b4cbc901/app/javascript/Admin/Components/NitroUI/Formik/Input.js)
- :green_circle: Green color signifies how that particular line is written in `neeto-ui` package
- :red_circle: Red color signifies how that particular line is written in `NitroUI` internals

## Sync Required or Not
- Are there significant changes for `formik/Input.js` that needs to be synced to `neeto-ui` package from `NitroUI`? - **No**

## Diff
```diff
--- NitroUI/formik/Input.js	2020-11-28 16:54:23.933435600 +0530
+++ neetoui-official/formik/Input.js	2020-11-28 16:49:18.769930700 +0530
@@ -6,12 +6,9 @@
 export default function FormikInput({ name, ...rest }) {
   return (
     <Field name={name}>
-      {({ field, form: { errors }, meta }) => {
-        const errorMessage = errors[field.name] || rest.error;
-        return (
-          <Input {...field} {...rest} error={meta.touched && errorMessage} />
-        );
-      }}
+      {({ field, meta }) => (
+        <Input error={meta.touched && meta.error} {...rest} {...field} />
+      )}
     </Field>
   );
 }
```

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #53 

@goutham-subramanyam _a

Please review.

### CODE REVIEW

File: lib/components/formik/Input.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ export default function FormikInput({ name, ...rest }) {
   return (
     <Field name={name}>
       {({ field, meta }) => (
-        <Input error={meta.touched && meta.error} {...rest} {...field} />
+        <Input {...field} {...rest} error={meta.touched && meta.error} />
```
Comment (goutham-subramanyam) [inline lib/components/formik/Input.js:10]:
```suggestion
        const errorMessage = errors[field.name] || rest.error;
        return (
          <Input {...field} {...rest} error={meta.touched && errorMessage} />
        );
```
It is like this now in NeetoHelp right?

**Review by goutham-subramanyam — COMMENTED** [review]
@yedhink _a

**Comment by yedhink** [comment]
@goutham-subramanyam _a

Let me know what you think about the new change. 

As per neeto-help version of this component, it only provided the user the ability to change the `error` message(not condition) by passing in as attr in  `rest` props.

With the current change:
- By default if no `error` attribute is passed as part of `rest`, then default formik error is used which get's invoked on being `touched`.
- Or if `error` attr is explicitly passed, then the invocation of the error condition is in full control of the user. The person can decide if the error needs to be shown during `touched` or whatever condition they prefer.

**Comment by goutham-subramanyam** [comment]
@yedhink _a 

> it only provided the user the ability to change the `error` message (not condition)

Yea that's a good improvement. My only concern is that the props were reordered by @danedavid because of some issue. So please check with him before updating the props order.

**Comment by yedhink** [comment]
@goutham-subramanyam _a

I have respected the re-ordering from neeto-help version. It was necessary in order to give precedence for attrs passed in via `rest` over `field`. 

With current changes that I have made, `rest` has the most priority.

**Comment by goutham-subramanyam** [comment]
@yedhink _a Okay. LGTM.

**Pull request merged**